domain/application/state/unit.go

```diff
@@ -1399,6 +1399,53 @@ func (st *State) GetUnitNamesForApplication(ctx context.Context, uuid coreapplic
 	}), nil
 }
 
+// GetUnitNamesForNetNode returns a slice of the unit names for the given net node
+// The following errors may be returned:
+func (st *State) GetUnitNamesForNetNode(ctx context.Context, uuid string) ([]coreunit.Name, error) {
+	db, err := st.DB()
+	if err != nil {
+		return nil, errors.Capture(err)
+	}
+
+	netNodeUUID := netNodeUUID{NetNodeUUID: uuid}
+	verifyExistsQuery := `SELECT COUNT(*) AS &countResult.count FROM net_node WHERE uuid = $netNodeUUID.uuid`
+	verifyExistsStmt, err := st.Prepare(verifyExistsQuery, countResult{}, netNodeUUID)
+	if err != nil {
+		return nil, errors.Capture(err)
+	}
```

Comment:
You should be able to get rid of this, for the moment it's not done, but every unit *must* have a net node, so if no rows are returned from the select, then the unit does not exist.

Reply:
I think a belts and braces approach here isn't such a bad thing. If you get an empty list of units because of a programmatic error, then I'd rather crash here...